types/module/core_services.go

```diff
@@ -0,0 +1,81 @@
+package module
+
+import (
+	"context"
+
+	"cosmossdk.io/core/store"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func NewCoreKVStoreService(key *sdk.KVStoreKey) store.KVStoreService {
```

Comment:
to use this we would need to break module apis?

Reply:
Well we're assuming the module is already using the core API

Reply:
makes sense thank you.